src/types/observablearray.ts

```diff
@@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
 		if (oldLength !== this.lastKnownLength)
 			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
 		this.lastKnownLength += delta;
-		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
-			reserveArrayBuffer(oldLength + delta);
+		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
+			reserveArrayBuffer(oldLength + delta + 1);
+		if (delta > 0)
```

Comment:
Basically, we can fix safari without it (I've added your commit because you said that you are going to merge it anyway).
I think, if we don't define own array props, all we need is to always define the only one prop, non-enumerable `0` (revert https://github.com/mobxjs/mobx/pull/396/commits/7565600e293c9a584fc3fd937e45131126f9d747).
So do we need making array props enumerable and loose performance?